packages/itwin/property-grid/src/components/PropertyGrid.tsx

```diff
@@ -171,6 +172,34 @@ export const PropertyGrid = ({
           await customOnDataChanged(dataProvider);
         }
       }
+      if(persistNullValueToggle) {
+        const res = await getShowNullValues();
```

Comment:
it seems like running this in the onDataChanged will cause it run more than it needs to be, it basically needs to be called just once then it only changes on user action

Reply:
I'm not sure where else to run it. It's async so I can't just run it by itself after the variable declarations. I could put it in a use callback of its own, but it depends on propertydata so it'd still run whenever the data is changed.

Reply:
it depends on the dataProvider, but does it need to run in ever onDataChanged event? This will lead to for example it running this code when you modify favorites etc

Reply:
True. I will move it to a callback dependent on dataProvider.

Reply:
Moved this to a useMemo that's just dependent on persistNullValueToggle.